rfcs/0000-go-generate.md

```diff
@@ -0,0 +1,177 @@
+# Go Generate Buildpack
+
+## Proposal
+The language family should have a `go generate` buildpack.
+
+<!---
+{{What changes are you purposing to the overall language family?}}
+-->
+
+## Motivation
+### Use Cases
+#### `go:generate` with a go tool
+
+##### How does this usually work in Dockerfile?
+
+#### `go:generate` with a tool installed on the system
+##### How does this work in a Dockerfile?
+
+#### `go:generate` with a binary generated during project build
+##### How does this work in a Dockerfile?
+
+<!---
+{{Why are we doing this? What pain points does this resolve? What use cases
+does it support? What is the expected outcome? Use real, concrete examples to
+make your case!}}
+-->
+
+## Implementation (Optional)
+### API
+If some file in the app source contains the directive
+```
+//go:generate some-tool
+```
+then the buildpack should require:
+```
+[[requires]]
+name = "some-tool"
+
+[[or]]
+
+[[or.requires]]
+mixin = "some-tool"
+```
+This takes advantage of the possibility of future
+[stackpacks](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md)
+providing the necessary tool.
+
+This buildpack should not provide anything.
+
+### Detection criteria
+The buildpack should pass detection if the files on which the generate command
+is run contain any  `//go:generate` directives AND a `BP_GO_GENERATE`
+environment variable is set to `true`. Its default value should be `true`.
+
+We can use `go generate -n ./...` (dry run) to search for directives.
```

Comment:
🤔  good point. That's unfortunate. We could get by without it but it would involve re-implementing the way `go generate` searches for directives.

Implementation would be simpler if detection did just depend on a `BP_GO_GENERATE` environment variable. Maybe that's okay if we think users will usually want lots of control over if/how `go generate` runs